### PR TITLE
Load `hook_msgs` when resume checkpoint

### PR DESCRIPTION
**Pull request by dreamerlin**
This PR is to load `hook_msgs` meta information saved in checkpoint which helps to store the `best` related information for evaluation

### CODE REVIEW

File: mmcv/runner/base_runner.py [hunk 1]
```diff
@@ -339,6 +339,11 @@ def resume(self,
 
         self._epoch = checkpoint['meta']['epoch']
         self._iter = checkpoint['meta']['iter']
+        if self.meta is None:
+            self.meta = dict()
+        self.meta.setdefault('hook_msgs', dict())
+        # load `last_ckpt`, `best_score`, `best_ckpt`, etc. for hook messages
+        self.meta['hook_msgs'].update(checkpoint['meta']['hook_msgs'])
```
Comment (hellock) [inline mmcv/runner/base_runner.py:346]: Does the key `hook_msgs` necessarily exist in `checkpoint['meta']`?

Comment (dreamerlin) [reply mmcv/runner/base_runner.py]: The key `hook_msgs` is used for keeping `best_score`, `best_ckpt`, etc. Which may be useful when resume checkpoint

File: mmcv/runner/base_runner.py [hunk 2]
```diff
@@ -339,6 +339,12 @@ def resume(self,
 
         self._epoch = checkpoint['meta']['epoch']
         self._iter = checkpoint['meta']['iter']
+        if self.meta is None:
+            self.meta = dict()
```
Comment (zhouzaida) [inline mmcv/runner/base_runner.py:343]: `self.meta = {}` may be more pythonic

File: mmcv/runner/base_runner.py [hunk 3]
```diff
@@ -339,6 +339,12 @@ def resume(self,
...
+        self.meta.setdefault('hook_msgs', dict())
```
Comment (zhouzaida) [inline mmcv/runner/base_runner.py:344]: self.meta.setdefault('hook_msgs', {})

File: mmcv/runner/base_runner.py (hunk 1 above)
Comment (hellock) [reply mmcv/runner/base_runner.py]: I mean this key may not exist in checkpoints of earlier versions.

File: mmcv/runner/hooks/evaluation.py [hunk 4]
```diff
@@ -292,13 +295,13 @@ class DistEvalHook(EvalHook):
...
         save_best (str, optional): If a metric is specified, it would measure
             the best checkpoint during evaluation. The information about best
-            checkpoint would be save in ``runner.meta['hook_msgs']``.
-            Options are the evaluation metrics to the test dataset. e.g.,
-            ``bbox_mAP``, ``segm_mAP`` for bbox detection and instance
-            segmentation. ``AR@100`` for proposal recall. If ``save_best`` is
-            ``auto``, the first key of the returned ``OrderedDict`` result
-            will be used. The interval of ``EvalHook`` should depend on
-            ``CheckpointHook``. Default: None.
+            checkpoint would be save in ``runner.meta['hook_msgs']`` to keep
```
Comment (zhouzaida) [inline mmcv/runner/hooks/evaluation.py:298]: save -> saved

File: mmcv/runner/hooks/evaluation.py [hunk 5]
```diff
@@ -29,13 +29,13 @@ class EvalHook(Hook):
...
-            Options are the evaluation metrics to the test dataset. e.g.,
-            ``bbox_mAP``, ``segm_mAP`` for bbox detection and instance
-            segmentation. ``AR@100`` for proposal recall. If ``save_best`` is
-            ``auto``, the first key of the returned ``OrderedDict`` result
-            will be used. The interval of ``EvalHook`` should be
-            divisible of that in ``CheckpointHook``. Default: None.
+            checkpoint would be saved in ``runner.meta['hook_msgs']`` to keep
+            best score value and best checkpoint path, which will be also
+            loaded when resume checkpoint. Options are the evaluation metrics
+            to the test dataset. e.g., ``bbox_mAP``, ``segm_mAP`` for bbox
```
Comment (ZwwWayne) [inline mmcv/runner/hooks/evaluation.py:35]: to the test dataset -> on the test dataset

**Review by ZwwWayne — APPROVED** [review]

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**